src/omv/modules/py_image.c

```diff
@@ -784,24 +797,78 @@ static mp_obj_t py_image_unpack(uint n_args, const mp_obj_t *pos_args, mp_map_t
         }
     }
 
-    if ((image->w * image->h * dtype_size * channels) > bufinfo.len) {
-        mp_raise_ValueError(MP_ERROR_TEXT("Buffer size is too small"));
+    ndarray_obj_t *ndarray;
+
+    if (args[ARG_buffer].u_obj != mp_const_none) {
+        mp_buffer_info_t bufinfo = {0};
+        mp_get_buffer_raise(args[ARG_buffer].u_obj, &bufinfo, MP_BUFFER_WRITE);
+
+        if ((len * dtype_size * channels) > bufinfo.len) {
+            mp_raise_ValueError(MP_ERROR_TEXT("Buffer is too small"));
+        }
+
+        ndarray = m_new_obj(ndarray_obj_t);
+        ndarray->base.type = &ulab_ndarray_type;
+        ndarray->dtype = dtype_code;
+        ndarray->boolean = NDARRAY_NUMERIC;
+        ndarray->ndim = ndim;
+        ndarray->len = len * channels;
+        ndarray->itemsize = dtype_size;
+        memcpy(ndarray->shape, shape, sizeof(shape));
+        memcpy(ndarray->strides, strides, sizeof(strides));
+        ndarray->array = bufinfo.buf;
+        ndarray->origin = bufinfo.buf;
```

Comment:
Will be hard to maintain this if internals change, maybe ask for a a new function that accepts the memory ?

Reply:
Yeah, that would help. They don't have one right now. I'll make a bug ticket for it.

https://github.com/v923z/micropython-ulab/issues/676